tests/unit/test_basics.py

```diff
@@ -269,3 +279,21 @@ def test_image_url(self):
             self.block.student_view_data()["target_img_expanded_url"],
             '/course/test-course/assets/foo.png',
         )
+
+    @ddt.data(
+        (False, now(), None, True),
+        (True, now(), None, False),
+        (False, now(), timedelta(days=1), False),
+        (True, now(), timedelta(days=1), False),
+        (False, now() - timedelta(hours=1), None, True),
+    )
+    @ddt.unpack
+    def test_submission_deadline(self, self_paced, due_date, graceperiod, is_submission_due):
```

Comment:
nit: to refactor `is_submission_due` to some other name as it seems a bit confusing at first glance.